bliss/surveys/dc2.py

```diff
@@ -0,0 +1,268 @@
+                tile_cat.to_dict()["star_log_fluxes"], "1 h w nh nw -> h w nh nw"
+            )
+
+            tile_dict["star_fluxes"] = tile_dict["star_fluxes"].clamp(min=1e-18)
+            tile_dict["galaxy_fluxes"] = tile_dict["galaxy_fluxes"].clamp(min=1e-18)
+            tile_dict["galaxy_params"][..., 3] = tile_dict["galaxy_params"][..., 3].clamp(min=1e-18)
+            tile_dict["galaxy_params"][..., 5] = tile_dict["galaxy_params"][..., 5].clamp(min=1e-18)
+
+            # split image to 25
+            split_h = np.linspace(0, plocs_lim[0], num=5, endpoint=False)
+            split_w = np.linspace(0, plocs_lim[1], num=5, endpoint=False)
+            image_h = plocs_lim[0] // 5
+            image_w = plocs_lim[1] // 5
+
+            for i in split_h:
+                for j in split_w:
+                    image1 = image[int(i) : int(i) + image_h, int(j) : int(j) + image_w]
+                    bg1 = bg_crop[int(i) : int(i) + image_h, int(j) : int(j) + image_w]
+
+                    tlim = [
+                        int(i // 4),
+                        int((i + image_h) // 4),
+                        int(j // 4),
+                        int((j + image_w) // 4),
+                    ]
+                    tile1 = get_tile((tlim[0], tlim[1], tlim[2], tlim[3]), tile_dict)
+
+                    data.append(
+                        {
+                            "tile_catalog": tile1,
+                            "images": rearrange(torch.from_numpy(image1), "h w -> 1 h w"),
+                            "background": rearrange(torch.from_numpy(bg1), "h w -> 1 h w"),
+                        }
+                    )
+
+        random.shuffle(data)
+        self.dc2_data = data
+
+    @property
+    def predict_batch(self):
+        if not self._predict_batch:
+            self._predict_batch = {
+                "images": self.dc2_data[0]["image"],
+                "background": self.dc2_data[0]["background"],
+            }
+        return self._predict_batch
+
+    @predict_batch.setter
+    def predict_batch(self, value):
+        self._predict_batch = value
+
+    def setup(self, stage="fit"):
+        data_len = len(self.dc2_data)
+        train_len = int(data_len * 0.8)
+        val_len = int(data_len * 0.1)
+
+        self.data = self.dc2_data[:train_len]
+        self.valid = self.dc2_data[train_len : train_len + val_len]
+        self.test = self.dc2_data[train_len + val_len :]
+
+    def train_dataloader(self):
+        return DataLoader(self.data, batch_size=1)
+
+    def val_dataloader(self):
+        return DataLoader(self.valid, batch_size=1)
+
+    def test_dataloader(self):
+        return DataLoader(self.test, batch_size=1)
+
+
+class Dc2FullCatalog(FullCatalog):
+    """Class for the LSST PHOTO Catalog.
+
+    Some resources:
+    - https://data.lsstdesc.org/doc/dc2_sim_sky_survey
+    - https://arxiv.org/abs/2010.05926
+    """
+
+    @classmethod
+    def load_dc2_catalog(cls, cat_path, band, wcs, plocs_lim):
```

Comment:
Is it possible to follow `FullCatalog::from_file` pattern here? See `SDSS::PhotoFullCatalog::from_file` and `DecalsFullCatalog::from_file`